### PR TITLE
Add Schneider Electric’s UMAS detection support

### PR DESCRIPTION
**Pull request by 0xA50C1A1**
Please sign (check) the below before submitting the Pull Request:

- [x] I have signed the ntop Contributor License Agreement at https://github.com/ntop/legal/blob/main/individual-contributor-licence-agreement.md
- [x] I have read the contributing guide lines at https://github.com/ntop/nDPI/blob/dev/CONTRIBUTING.md
- [x] I have updated the documentation (in doc/) to reflect the changes made (if applicable)

Describe changes:

UMAS is a Modbus-based proprietary protocol (or extension?) from Schneider Electric, used in their PLCs. Huh, adding support for this turned out to be quite trivial, as SE reserved a special Modbus function code for UMAS.

### CODE REVIEW

File: src/lib/protocols/modbus.c [hunk 1]
```diff
@@ -47,9 +47,16 @@ static void ndpi_search_modbus_tcp(struct ndpi_detection_module_struct *ndpi_str
 
       // the fourth parameter of the payload is the length of the segment            
       if((modbus_len-1) == (packet->payload_packet_len - 7 /* ModbusTCP header len */)) {
-	NDPI_LOG_INFO(ndpi_struct, "found MODBUS\n");
-	ndpi_set_detected_protocol(ndpi_struct, flow, NDPI_PROTOCOL_MODBUS, NDPI_PROTOCOL_UNKNOWN, NDPI_CONFIDENCE_DPI);
-	return;
+        /* Check Modbus function code. 0x5A (90) is reserved for UMAS protocol */
+        if (packet->payload[7] == 0x5A) {
+          NDPI_LOG_INFO(ndpi_struct, "found Schneider Electric UMAS\n");
+          ndpi_set_detected_protocol(ndpi_struct, flow, NDPI_PROTOCOL_MODBUS, NDPI_PROTOCOL_UMAS, NDPI_CONFIDENCE_DPI);
```
Comment (IvanNardi) [inline src/lib/protocols/modbus.c]: The other way around: `ndpi_struct, flow, NDPI_PROTOCOL_UMAS, NDPI_PROTOCOL_MODBUS, ...`

**Review by IvanNardi — CHANGES_REQUESTED** [review]
LGTM, just a small issue

**Review by IvanNardi — APPROVED** [review]

**Comment by IvanNardi** [comment]
@utoni [unrelated to this PR] could you take a look at the error  when downloading the gambling sites list, please? The HTML page might be different now and we might need to update the `sed` command...

**Comment by IvanNardi** [comment]
Thanks!

**Comment by utoni** [comment]
> @utoni [unrelated to this PR] could you take a look at the error when downloading the gambling sites list, please? The HTML page might be different now and we might need to update the `sed` command...

will do

**Comment by IvanNardi** [comment]
> > @utoni [unrelated to this PR] could you take a look at the error when downloading the gambling sites list, please? The HTML page might be different now and we might need to update the `sed` command...
> 
> will do

Thanks!

**Pull request merged**